yt/yt/server/lib/state_checker/state_checker.cpp

```diff
@@ -0,0 +1,88 @@
+#include "private.h"
+#include "state_checker.h"
+
+#include <yt/yt/core/logging/log.h>
+
+#include <yt/yt/core/concurrency/periodic_executor.h>
+
+namespace NYT::NStateChecker {
+
+using namespace NApi;
+using namespace NConcurrency;
+using namespace NLogging;
+using namespace NYson;
+using namespace NYTree;
+
+////////////////////////////////////////////////////////////////////////////////
+
+TStateChecker::TStateChecker(IInvokerPtr invoker, IClientPtr nativeClient, TYPath instancePath, TDuration stateCheckPeriod)
+    : Logger(StateCheckerLogger())
+    , Invoker_(std::move(invoker))
+    , NativeClient_(std::move(nativeClient))
+    , InstancePath_(std::move(instancePath))
+{
+    Banned_.store(false);
+    StateCheckerExecutor_ = New<TPeriodicExecutor>(
+        Invoker_,
+        BIND(&TStateChecker::DoCheckState, MakeStrong(this)),
+        stateCheckPeriod);
+}
+
+void TStateChecker::Start()
+{
+    StateCheckerExecutor_->Start();
+}
+
+void TStateChecker::SetPeriod(TDuration stateCheckPeriod)
+{
+    StateCheckerExecutor_->SetPeriod(stateCheckPeriod);
+}
+
+bool TStateChecker::IsComponentBanned() const
+{
+    return Banned_.load();
+}
+
+////////////////////////////////////////////////////////////////////////////////
+
+void TStateChecker::DoCheckState()
+{
+    VERIFY_INVOKER_AFFINITY(Invoker_);
+
+    YT_LOG_DEBUG("Started checking component state");
+    auto logFinally = Finally([&] {
+        YT_LOG_DEBUG("Finished checking component state (Banned: %v)", Banned_.load());
+    });
+
+    auto options = TGetNodeOptions{
+        .Attributes = TAttributeFilter({BannedAttributeName}),
+    };
+
+    try {
+        auto yson = WaitFor(NativeClient_->GetNode(InstancePath_, options))
+            .ValueOrThrow();
+        auto instance = ConvertToNode(yson);
+
+        auto banned = false;
+        if (instance->Attributes().Contains(BannedAttributeName)) {
+            banned = instance->Attributes().Get<bool>(BannedAttributeName);
+        }
+
+        auto previousBanned = Banned_.exchange(banned);
+        if (previousBanned != banned) {
+            auto orchidBannedPath = InstancePath_ + "/orchid/@banned";
+
+            YT_LOG_DEBUG("Updating orchid node (Path: %v, Banned: %v)", orchidBannedPath, banned);
+            WaitFor(NativeClient_->SetNode(orchidBannedPath, ConvertToYsonString(banned)))
+                .ThrowOnError();
+            YT_LOG_DEBUG("Orchid node updated");
+        }
```

Comment:
Why do we need to do it? Seems like `orchid` should belong to a node and should be managed by a node.
Also we already have this attribute on `instance/@banned`, I don't think we need an additional one on `instance/orchid/@banned`

Reply:
It was made to reduce sleeping in tests.
https://github.com/ytsaurus/ytsaurus/pull/851#discussion_r1763027352

This attribute is different. It means that component already knows about ban and won't continue to work.

Reply:
I see, it is used only in yql agent and query tracker itself, so they are still responsible for their `orchid`

You said in some reply:
> In some moment proxies will call IsUp function in qt - which tells, that it stops working - and stops sending queries to it.

But isUp exists only in StateChecker which should exist only in yql agent and query tracker
Also I don't see any IsUp usages in the PR. Could you please help me find them?

Reply:
It is a common RPC mechanic in Discover requests
https://github.com/ytsaurus/ytsaurus/blob/main/yt/yt/core/rpc/service_detail.cpp#L2735

Reply:
I see. My misunderstanding was here:
> isUp exists only in StateChecker

isUp is a separate API that forwards the request to StateChecker

Thanks